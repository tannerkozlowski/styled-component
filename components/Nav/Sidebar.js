import styled, { css } from 'styled-components'

import rem from '../../utils/rem'
import { mobile } from '../../utils/media'
import { paleGrey } from '../../utils/colors'
import { headerFont } from '../../utils/fonts'
import captureScroll from '../CaptureScroll'

const Sidebar = styled.nav`
  position: fixed;
  transform: translateZ(0);
  display: block;
  z-index: 5;
  font-family: ${headerFont};

  left: 0;
  top: 0;
  bottom: 0;
  right: auto;

  width: ${rem(300)};
  height: 100%;
  background: ${paleGrey};
  box-sizing: border-box;
  color: inherit;
  overflow-y: scroll;
  transition: transform 150ms ease-out;

  ${mobile(css`
    ${p => p.isFolded ? css`
      transform: translateX(${rem(-300)});
    `: ``}
  `)}
`

export default captureScroll(Sidebar)
