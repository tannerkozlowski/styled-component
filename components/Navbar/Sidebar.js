import styled, { css } from 'styled-components'
import rem from 'polished/lib/helpers/rem'
import { mobile } from '../../utils/media'

const Sidebar = styled.nav`
  position: fixed;
  display: block;
  z-index: 1;

  left: 0;
  top: 0;
  bottom: 0;
  right: auto;

  width: ${rem(300)};
  height: 100%;
  background: linear-gradient(20deg, rgb(219, 112, 147), rgb(243, 182, 97));
  box-sizing: border-box;
  color: white;
  overflow-y: scroll;

  ${mobile(css`
    bottom: auto;
    right: 0;
    height: auto;
    width: 100%;
    overflow-y: visible;
  `)}
`

export default Sidebar
