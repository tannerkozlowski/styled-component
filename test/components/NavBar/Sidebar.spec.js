import React from 'react'
import { mount } from 'enzyme'
import 'jest-styled-components'
import Sidebar from '../../../components/Nav/Sidebar'

test('Sidebar renders correctly', () => {
  const wrapper = mount(<Sidebar />)

  expect(wrapper).toMatchSnapshot()
})
