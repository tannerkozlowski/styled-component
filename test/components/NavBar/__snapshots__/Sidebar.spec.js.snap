// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar renders correctly 1`] = `
.c0 {
  position: fixed;
  -webkit-transform: translateZ(0);
  -ms-transform: translateZ(0);
  transform: translateZ(0);
  display: block;
  z-index: 5;
  font-family: "Avenir Next",-apple-system,BlinkMacSystemFont,"Segoe UI",Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  left: 0;
  top: 0;
  bottom: 0;
  right: auto;
  width: 16.666666666666668rem;
  height: 100%;
  background: #efefef;
  box-sizing: border-box;
  color: inherit;
  overflow-y: scroll;
  -webkit-transition: -webkit-transform 150ms ease-out;
  -webkit-transition: transform 150ms ease-out;
  transition: transform 150ms ease-out;
}

<styled.nav>
  <styled.nav>
    <nav
      className="c0"
    />
  </styled.nav>
</styled.nav>
`;
